mcorpc/authz_rego.go

```diff
@@ -45,39 +43,26 @@ func (r *regoPolicy) authorize() (bool, error) {
 		return false, err
 	}
 
-	pol, err := ioutil.ReadFile(policyFile)
-	if err != nil {
-		return false, err
-	}
-
-	module := string(pol)
-
-	buf := topdown.NewBufferTracer()
-
-	if r.log.Logger.GetLevel() == logrus.DebugLevel {
-		r.log.Debugf("regoInputs: %v", r.regoInputs())
+	if policyFile == "" {
+		return false, fmt.Errorf("policy file could not be found")
 	}
 
-	options := []func(*rego.Rego){
-		rego.Query("data.choria.mcorpc.authpolicy.allow"),
-		rego.Module(policyFile, module),
-		rego.Input(r.regoInputs()),
+	eopts := []opa.Option{
+		opa.Logger(r.log),
+		opa.File(policyFile),
 	}
 
-	if (r.log.Logger.GetLevel() == logrus.DebugLevel) || r.enableTracing() {
-		options = append(options, rego.Tracer(buf))
+	if r.log.Logger.GetLevel() == logrus.DebugLevel || r.enableTracing() {
+		r.log.Debugf("regoInputs: %v", r.regoInputs())
```

Comment:
i didnt put this behaviour in go-security, didnt want to assume that always the contents of inputs is safe to go into log files, ok here as its on the node in question anyway